config/configmiddleware/README.md

```diff
@@ -0,0 +1,37 @@
+# OpenTelemetry Collector Middleware Configuration
+
+This package implements a configuration struct for referring to
+[middleware extensions](../../extension/extensionmiddleware/README.md).
+
+## Overview
+
+The `configmiddleware` package defines a `Middleware` type that
+allows components to configure middleware extensions, typically as
+an ordered list.
+This support is built in for push-based receivers configured through
+`confighttp` and `configgrpc`, as for example in the OTLP receiver:
+
+```yaml
+receivers:
+  otlp:
+    protocols:
+	  http:
+	    middlewares:
+		- middleware: limitermiddleware
```

Comment:
Looking at this config, should we replace `middleware` with `id` within the `Middleware` struct?

Reply:
I copied configauth in this, which uses 
	AuthenticatorID component.ID `mapstructure:"authenticator,omitempty"`